diff --git a/companion/src/CMakeLists.txt b/companion/src/CMakeLists.txt
--- a/companion/src/CMakeLists.txt
+++ b/companion/src/CMakeLists.txt
@@ -1,7 +1,7 @@
 set(SIMULATOR_INSTALL_PREFIX "" CACHE STRING "Alternative simulator library search path")
 
-set(COMPANION_NAME "companion")
-set(SIMULATOR_NAME "simulator")
+set(COMPANION_NAME "edgetx-companion")
+set(SIMULATOR_NAME "edgetx-simulator")
 set(APP_NAME_SUFFIX "")
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "(Linux|FreeBSD)")
@@ -15,7 +15,7 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "(Linux|FreeBSD)")
   if (NOT ${SIMULATOR_INSTALL_PREFIX} STREQUAL "")
     set(SIMULATOR_LIB_PATH ${SIMULATOR_INSTALL_PREFIX}/lib/${COMPANION_NAME})
   else()
-    set(SIMULATOR_LIB_PATH "../lib/${COMPANION_NAME}")
+    set(SIMULATOR_LIB_PATH ${LIB_INSTALL_DIR}/${COMPANION_NAME}-${APP_NAME_SUFFIX})
   endif()
   message(STATUS "Simulators library search path: " ${SIMULATOR_LIB_PATH})
 endif()
@@ -334,7 +334,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
 
   # libsim*.so files
   install(DIRECTORY ${CMAKE_BINARY_DIR}/plugins/
-          DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/${COMPANION_NAME})
+          DESTINATION ${SIMULATOR_LIB_PATH})
 
   # application icons
   set(icon_sizes "16;22;24;32;48;64;128;256;512")
