diff --git a/companion/targets/linux/companion.desktop.in b/companion/targets/linux/companion.desktop.in
--- a/companion/targets/linux/companion.desktop.in
+++ b/companion/targets/linux/companion.desktop.in
@@ -7,4 +7,4 @@ Icon=@COMPANION_NAME@
 Exec=@COMPANION_NAME@
 Terminal=false
 StartupNotify=false
-Categories=Utility;
+Categories=Utility;Sports;Electronics;Emulator;Robotics;Qt;Science
diff --git a/companion/targets/linux/simulator.desktop.in b/companion/targets/linux/simulator.desktop.in
index a86cb95765..f4cc88b3f1 100644
--- a/companion/targets/linux/simulator.desktop.in
+++ b/companion/targets/linux/simulator.desktop.in
@@ -7,4 +7,4 @@ Icon=@COMPANION_NAME@
 Exec=@SIMULATOR_NAME@
 Terminal=false
 StartupNotify=false
-Categories=Utility;
+Categories=Utility;Sports;Electronics;Emulator;Robotics;Qt;Science
