diff --git a/companion/src/CMakeLists.txt b/companion/src/CMakeLists.txt
--- a/companion/src/CMakeLists.txt
+++ b/companion/src/CMakeLists.txt
@@ -569,56 +569,6 @@ elseif(APPLE)
       file(RENAME \"\${app_dir}\" \"\${CMAKE_INSTALL_PREFIX}/${COMPANION_OSX_APP_BUNDLE_NAME}.app\")
   " COMPONENT Runtime)
 
-else()
-  set(LINUXDEPLOY_APPIMAGE "linuxdeploy-x86_64.AppImage")
-  set(LINUXDEPLOY_PLUGIN_QT "linuxdeploy-plugin-qt-x86_64.AppImage")
-  set(LINUXDEPLOY_URL "https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous")
-  set(LINUXDEPLOY_PLUGIN_QT_URL "https://github.com/linuxdeploy/linuxdeploy-plugin-qt/releases/download/continuous")
-  set(LINUXDEPLOY_DIRECTORY "${CMAKE_BINARY_DIR}/linuxdeploy")
-  set(LINUXDEPLOY_APP "${LINUXDEPLOY_DIRECTORY}/usr/bin/linuxdeploy")
-
-  if(NOT EXISTS "${LINUXDEPLOY_DIRECTORY}")
-    message(STATUS "Downloading linuxdeploy and plugins...")
-
-    file(DOWNLOAD "${LINUXDEPLOY_URL}/${LINUXDEPLOY_APPIMAGE}" "${CMAKE_BINARY_DIR}/${LINUXDEPLOY_APPIMAGE}")
-    file(DOWNLOAD "${LINUXDEPLOY_PLUGIN_QT_URL}/${LINUXDEPLOY_PLUGIN_QT}" "${CMAKE_BINARY_DIR}/${LINUXDEPLOY_PLUGIN_QT}")
-
-    message(STATUS "Downloading linuxdeploy and plugins finished")
-
-    # for security reasons by default AppImages are prohibited from running in Docker containers
-    # so extract Appimage contents and run as normal apps
-
-    file(GLOB linuxdeployfiles ${CMAKE_BINARY_DIR}/linuxdeploy*.AppImage)
-
-    foreach(file ${linuxdeployfiles})
-      execute_process(
-        COMMAND chmod +x "${file}"
-        WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-
-      execute_process(
-        COMMAND "${file}" --appimage-extract
-        WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-
-      file(REMOVE "${file}")
-    endforeach()
-
-    file(REMOVE_RECURSE "${LINUXDEPLOY_DIRECTORY}")
-
-    execute_process(
-      COMMAND mv "${CMAKE_BINARY_DIR}/squashfs-root" "${LINUXDEPLOY_DIRECTORY}"
-      WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-
-    # test installation
-    execute_process(
-      COMMAND ${LINUXDEPLOY_APP} --list-plugins
-      WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-  endif()
-
-  list(APPEND CPACK_GENERATOR "External")
-  set(CPACK_PACKAGE_DIRECTORY "${CMAKE_BINARY_DIR}")
-  set(APPIMAGE_DIR "${CPACK_PACKAGE_DIRECTORY}/_CPack_Packages/Linux/External/AppImage")
-  configure_file("${COMPANION_TARGETS_DIR}/CPackLinuxDeploy.cmake.in" "${CMAKE_BINARY_DIR}/CPackExternal.cmake" @ONLY)
-  set(CPACK_EXTERNAL_PACKAGE_SCRIPT "${CMAKE_BINARY_DIR}/CPackExternal.cmake")
 endif()
 
 include(CPack)
