diff --git a/companion/src/CMakeLists.txt b/companion/src/CMakeLists.txt
--- a/companion/src/CMakeLists.txt
+++ b/companion/src/CMakeLists.txt
@@ -6,8 +6,8 @@ set(SIMULATOR_NAME "simulator")
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "(Linux|FreeBSD)")
   set(APP_NAME_SUFFIX "${VERSION_MAJOR}${VERSION_MINOR}")
-  set(COMPANION_NAME "${COMPANION_NAME}${APP_NAME_SUFFIX}")
-  set(SIMULATOR_NAME "${SIMULATOR_NAME}${APP_NAME_SUFFIX}")
+  set(COMPANION_NAME "edgetx-companion")
+  set(SIMULATOR_NAME "edgetx-simulator")
   # by default cmake sets CMAKE_INSTALL_PREFIX to /usr/local however linuxdeploy works best with '/usr' as base directory
   if (${CMAKE_INSTALL_PREFIX} MATCHES "/usr/local")
     set(CMAKE_INSTALL_PREFIX "/usr")
@@ -15,7 +15,7 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "(Linux|FreeBSD)")
   if (NOT ${SIMULATOR_INSTALL_PREFIX} STREQUAL "")
     set(SIMULATOR_LIB_PATH ${SIMULATOR_INSTALL_PREFIX}/lib/${COMPANION_NAME})
   else()
-    set(SIMULATOR_LIB_PATH "../lib/${COMPANION_NAME}")
+    set(SIMULATOR_LIB_PATH ${LIB_INSTALL_DIR}/${COMPANION_NAME})
   endif()
   message(STATUS "Simulators library search path: " ${SIMULATOR_LIB_PATH})
 endif()
@@ -194,8 +194,8 @@ set(CPN_COMMON_LIB common)
 
 ############# Supporting libraries ###############
 
-include(FetchMiniz)
-include(FetchYamlCpp)
+find_package(miniz CONFIG)
+find_package(yaml-cpp REQUIRED CONFIG)
 include(FetchMaxLibQt)
 
 add_subdirectory(datamodels)
@@ -328,13 +328,15 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   # configure and set variables used by package script
   set(COMPANION_DESKTOP_FILE ${CMAKE_CURRENT_BINARY_DIR}/${COMPANION_NAME}.desktop)
   configure_file(${COMPANION_TARGETS_DIR}/companion.desktop.in ${COMPANION_DESKTOP_FILE} @ONLY)
+  install(FILES ${COMPANION_DESKTOP_FILE} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications)
 
   set(SIMULATOR_DESKTOP_FILE ${CMAKE_CURRENT_BINARY_DIR}/${SIMULATOR_NAME}.desktop)
   configure_file(${COMPANION_TARGETS_DIR}/simulator.desktop.in ${SIMULATOR_DESKTOP_FILE} @ONLY)
+  install(FILES ${SIMULATOR_DESKTOP_FILE} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications)
 
   # libsim*.so files
   install(DIRECTORY ${CMAKE_BINARY_DIR}/plugins/
-          DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/${COMPANION_NAME})
+          DESTINATION ${SIMULATOR_LIB_PATH})
 
   # application icons
   set(icon_sizes "16;22;24;32;48;64;128;256;512")
@@ -351,15 +353,6 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
           DESTINATION "${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/apps"
           RENAME "${COMPANION_NAME}.svg")
 
-  if(OPENSSL_FOUND)
-    get_filename_component(OPENSSL_SSL_LIBRARY_DIR ${OPENSSL_SSL_LIBRARY} DIRECTORY)
-    install(DIRECTORY ${OPENSSL_SSL_LIBRARY_DIR}/
-            DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
-            FILES_MATCHING
-              PATTERN "libcrypto.so.*"
-              PATTERN "libssl.so.*")
-  endif()
-
   set(OperatingSystem "Linux")
   # Shortcut target
   add_custom_target(companion DEPENDS ${COMPANION_NAME})
