diff --git a/companion/src/helpers.cpp b/companion/src/helpers.cpp
index e895d9c29..cf49af65c 100644
--- a/companion/src/helpers.cpp
+++ b/companion/src/helpers.cpp
@@ -422,7 +422,7 @@ void startSimulation(QWidget * parent, RadioData & radioData, int modelIdx)
 #elif defined Q_OS_APPLE
   const QString program = "simulator.dmg";
 #else
-  const QString program = QString("simulator%1%2").arg(VERSION_MAJOR).arg(VERSION_MINOR);
+  const QString program = QString("edgetx-simulator");
 #endif
 
   QStringList arguments;
