diff --git a/companion/src/CMakeLists.txt b/companion/src/CMakeLists.txt
--- a/companion/src/CMakeLists.txt
+++ b/companion/src/CMakeLists.txt
@@ -552,39 +552,4 @@ if(APPLE)
   set(CPACK_PACKAGE_FILE_NAME "edgetx-${CPACK_PACKAGE_NAME_LOWERCASE}-${VERSION}")
 endif(APPLE)
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-  get_filename_component(_qt_bin_dir "${QT_QMAKE_EXECUTABLE}" DIRECTORY)
-  find_program(LINUXDEPLOY_EXECUTABLE linuxdeployqt HINTS "${_qt_bin_dir}")
-
-  if(NOT LINUXDEPLOY_EXECUTABLE)
-    message(STATUS "linuxdeployqt will be downloaded")
-
-    set(LINUXDEPLOY_EXECUTABLE "${CMAKE_BINARY_DIR}/deployqt/AppRun")
-    set(LINUXDEPLOY_URL "https://github.com/probonopd/linuxdeployqt/releases/download/continuous/linuxdeployqt-continuous-x86_64.AppImage")
-    set(LINUXDEPLOY_BUNDLE "${CMAKE_BINARY_DIR}/linuxdeployqt.AppImage")
-
-    if (NOT EXISTS "${LINUXDEPLOY_EXECUTABLE}")
-      file(DOWNLOAD "${LINUXDEPLOY_URL}" "${LINUXDEPLOY_BUNDLE}")
-      execute_process(
-        COMMAND chmod +x "${LINUXDEPLOY_BUNDLE}"
-        WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-      execute_process(
-        COMMAND "${LINUXDEPLOY_BUNDLE}" --appimage-extract
-        WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-
-      file(REMOVE_RECURSE "${CMAKE_BINARY_DIR}/deployqt")
-      execute_process(
-        COMMAND mv "${CMAKE_BINARY_DIR}/squashfs-root" deployqt
-        WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-      file(REMOVE "${LINUXDEPLOY_BUNDLE}")
-    endif()
-  endif()
-
-  set(CPACK_PACKAGE_DIRECTORY "${CMAKE_BINARY_DIR}")
-  set(CPACK_GENERATOR "External")
-
-  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/CPackLinuxDeployQt.cmake.in "${CMAKE_BINARY_DIR}/CPackExternal.cmake" @ONLY)
-  set(CPACK_EXTERNAL_PACKAGE_SCRIPT "${CMAKE_BINARY_DIR}/CPackExternal.cmake")
-endif()
-
 include(CPack)
