diff --git a/companion/src/companion.desktop.in b/companion/src/companion.desktop.in
--- a/companion/src/companion.desktop.in
+++ b/companion/src/companion.desktop.in
@@ -7,4 +7,4 @@
 Exec=@COMPANION_NAME@
 Terminal=false
 StartupNotify=false
-Categories=Utility;
+Categories=Utility;Sports;Electronics;Emulator;Robotics;Qt;Science
diff --git a/companion/src/simulator.desktop.in b/companion/src/simulator.desktop.in
--- a/companion/src/simulator.desktop.in
+++ b/companion/src/simulator.desktop.in
@@ -7,4 +7,4 @@
 Exec=@SIMULATOR_NAME@
 Terminal=false
 StartupNotify=false
-Categories=Utility;
+Categories=Utility;Sports;Electronics;Emulator;Robotics;Qt;Science
