diff --git a/cmake/GenericDefinitions.cmake b/cmake/GenericDefinitions.cmake
index 5f97f15ba6..95b1ad6fec 100644
--- a/cmake/GenericDefinitions.cmake
+++ b/cmake/GenericDefinitions.cmake
@@ -2,10 +2,6 @@
 # https://cmake.org/cmake/help/latest/policy/CMP0020.html
 cmake_policy(SET CMP0020 NEW)
 
-# Allow keyword and plain target_link_libraries() signatures to be mixed
-# https://cmake.org/cmake/help/latest/policy/CMP0023.html
-cmake_policy(SET CMP0023 OLD)
-
 # Use @rpath in a target's install name on MacOS X for locating shared libraries
 # https://cmake.org/cmake/help/latest/policy/CMP0042.html
 if(POLICY CMP0042)
diff --git a/companion/src/CMakeLists.txt b/companion/src/CMakeLists.txt
index 8cc500e7a6..4971d65ecf 100644
--- a/companion/src/CMakeLists.txt
+++ b/companion/src/CMakeLists.txt
@@ -344,9 +344,6 @@ elseif(PCB STREQUAL X10 AND PCBREV STREQUAL T18)
 else()
   string(TOLOWER ${PCB} FLAVOUR)
 endif()
-if(POLICY CMP0026)
-  cmake_policy(SET CMP0026 OLD)  # https://cmake.org/cmake/help/v3.0/policy/CMP0026.html
-endif()
 
 # OpenSSL
 # environment variable set in github workflows and build-edgetx Dockerfile
