diff --git a/companion/targets/linux/companion.desktop.in b/companion/targets/linux/companion.desktop.in
--- a/companion/targets/linux/companion.desktop.in
+++ b/companion/targets/linux/companion.desktop.in
@@ -8,4 +8,4 @@ Exec=@COMPANION_NAME@
 Terminal=false
 StartupWMClass=EdgeTX Companion
 StartupNotify=false
-Categories=Utility;
+Categories=Utility;Sports;Electronics;Emulator;Robotics;Qt;Science
diff --git a/companion/targets/linux/simulator.desktop.in b/companion/targets/linux/simulator.desktop.in
--- a/companion/targets/linux/simulator.desktop.in
+++ b/companion/targets/linux/simulator.desktop.in
@@ -7,4 +7,4 @@ Icon=@COMPANION_NAME@
 Exec=@SIMULATOR_NAME@
 Terminal=false
 StartupNotify=false
-Categories=Utility;
+Categories=Utility;Sports;Electronics;Emulator;Robotics;Qt;Science
