diff --git a/radio/src/CMakeLists.txt b/radio/src/CMakeLists.txt
index 20852450c..99c823c01 100644
--- a/radio/src/CMakeLists.txt
+++ b/radio/src/CMakeLists.txt
@@ -504,7 +504,7 @@ if(NOT MSVC)
     elseif(OPENRC_BOOTLOADER)
       # We fetch Mike's bootloader as it is not included inside OpenTX
       message(STATUS "Downloading Mike's bootloader: https://github.com/MikeBland/OpenRcBootloader/releases/download/V1.9/${OPENRC_BOOTLOADER}")
-      file(DOWNLOAD https://github.com/MikeBland/OpenRcBootloader/releases/download/V1.9/${OPENRC_BOOTLOADER} ${CMAKE_CURRENT_BINARY_DIR}/bootloader.lbm STATUS status)
+      file(CREATE_LINK ${CMAKE_CURRENT_BINARY_DIR}/${OPENRC_BOOTLOADER} ${CMAKE_CURRENT_BINARY_DIR}/bootloader.lbm COPY_ON_ERROR)
       list(GET status 0 error_code)
       if(error_code)
         list(GET status 1 error_message)
